sdk/src/beta9/channel.py

```diff
@@ -18,27 +18,58 @@
     ConfigContext,
     SDKSettings,
     get_config_context,
+    get_settings,
     load_config,
     prompt_for_config_context,
     save_config,
 )
 from .exceptions import RunnerException
 
 
+def channel_reconnect_event(connect_status: grpc.ChannelConnectivity) -> None:
+    if connect_status not in (
+        grpc.ChannelConnectivity.CONNECTING,
+        grpc.ChannelConnectivity.IDLE,
+        grpc.ChannelConnectivity.READY,
+        grpc.ChannelConnectivity.SHUTDOWN,
+    ):
+        terminal.warn("Connection lost, reconnecting...")
+
+
+def get_user_agent() -> str:
+    from importlib import metadata
+
+    package_name = get_settings().name.lower()
+    return f"{package_name}/{metadata.version(package_name)}"
+
+
 class Channel(InterceptorChannel):
     def __init__(
         self,
         addr: str,
         token: Optional[str] = None,
         credentials: Optional[ChannelCredentials] = None,
+        options: Optional[Sequence[Tuple[str, Any]]] = None,
+        retry: Tuple[Callable[[grpc.ChannelConnectivity], None], bool] = (
+            channel_reconnect_event,
+            True,
+        ),
     ):
+        if options is None:
+            options = []
+
+        options = [opt for opt in options if "grpc.secondary_user_agent" not in opt[0]]
+        options.append(("grpc.secondary_user_agent", get_user_agent()))
+
         if credentials is not None:
             channel = grpc.secure_channel(addr, credentials)
         elif addr.endswith("443"):
             channel = grpc.secure_channel(addr, grpc.ssl_channel_credentials())
         else:
             channel = grpc.insecure_channel(addr)
 
+        channel.subscribe(*retry)
```

Comment:
Setting this to True (the default) lets the channel reconnect on disconnects. It works really well from my testing.